adb/filesync_protocol.py

```diff
@@ -211,7 +210,13 @@ def Read(self, expected_ids, read_data=True):
 
     if command_id not in expected_ids:
       if command_id == b'FAIL':
-        raise usb_exceptions.AdbCommandFailureException('Command failed.')
+        reason = ''
+        if self.recv_buffer:
+            try:
+                reason = self.recv_buffer.decode('utf-8')
+            except:
+                pass
+        raise usb_exceptions.AdbCommandFailureException('Command failed: {}'.format(reason))
```

Comment:
there's an easier way to decode potentially unknown data:
`reason = self.recv_buffer.decode('utf8', 'ignore')`
https://docs.python.org/2.7/library/codecs.html#codec-base-classes

Reply:
Fixed